### PR TITLE
Enhancement: Add serviceAccountUser permission when terraform is running as a service-account

### PR DESCRIPTION
**Pull request by lpereir4**
## Description

The `iam.serviceAccountUser` role is necessary when deploying a cloud-run service or cloud-scheduler job.
Granting this at project level is very broad and is [considered unsafe](https://cloud.google.com/security-command-center/docs/how-to-remediate-security-health-analytics-findings#over_privileged_service_account_user).
When terraform is running as a service-account the module will explicitly use the `iam.serviceAccountUser` role on the invoker and cleaner SAs to the account that is running terraform.

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [x] My PR title follows [semantics prefix](.github/CONTRIBUTING.md#semantic-pull-requests)
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked my code and corrected any misspellings
- [x] I have run pre-commit hooks `pre-commit run -a`
- [x] CI tests are passing

### CODE REVIEW

File: iam.tf [hunk 1]
```diff
@@ -52,3 +52,23 @@ resource "google_artifact_registry_repository_iam_member" "this" {
 
   provider = google-beta
 }
+
+# Allow the account that is running terraform permissions to act-as
+# the service-account, required for deploying a CloudRun service
+resource "google_service_account_iam_member" "tf_cleaner" {
```
Comment (anouarchattouna) [inline iam.tf:58]:
```suggestion
resource "google_service_account_iam_member" "tf_as_cleaner" {
```

File: iam.tf [hunk 2]
```diff
@@ -52,3 +52,23 @@ resource "google_artifact_registry_repository_iam_member" "this" {
...
+  count = local.running_as_a_service_account ? 1 : 0
+
+  service_account_id = google_service_account.cleaner.name
+  role               = "roles/iam.serviceAccountUser"
+  member             = "serviceAccount:${data.google_client_openid_userinfo.terraform.email}"
+}
+
+# Allow the account that is running terraform permissions to act-as
+# the service-account, required for deploying a CloudScheduler service
+resource "google_service_account_iam_member" "tf_invoker" {
```
Comment (anouarchattouna) [inline iam.tf:68]:
```suggestion
resource "google_service_account_iam_member" "tf_as_invoker" {
```

File: data.tf [hunk 3]
```diff
@@ -6,3 +6,5 @@ data "google_storage_bucket" "bucket" {
 
   name = each.value
 }
+
+data "google_client_openid_userinfo" "terraform" {}
```
Comment (anouarchattouna) [inline data.tf:10]:
```suggestion
# Get OpenID userinfo about the credentials used with the Google provider, specifically the email of the user applying the terrafom plan
data "google_client_openid_userinfo" "terraform" {}
```

**Review by anouarchattouna — APPROVED** [review]
Small modifications, otherwise LGTM!

**Review by grandria — APPROVED** [review]

**Pull request merged**